lib/c-api/src/wasm_c_api/store.rs

```diff
@@ -1,11 +1,11 @@
 use super::engine::wasm_engine_t;
 use std::cell::UnsafeCell;
-use std::sync::Arc;
+use std::rc::Rc;
 use wasmer_api::{AsStoreMut, AsStoreRef, Store, StoreMut, StoreRef as BaseStoreRef};
 
 #[derive(Clone)]
 pub struct StoreRef {
-    inner: Arc<UnsafeCell<Store>>,
```

Comment:
Why this change?
This would make the StoreRef non-transferable between threads.

Reply:
We are having an Arc over an element that is not `Send` and `Sync`, which basically triggers a compilation error in latest Rust @theduke